packages/expo-google-sign-in/android/src/main/java/expo/modules/google/signin/GoogleSignInModule.java

```diff
@@ -0,0 +1,449 @@
+        });
+    }
+
+    @ExpoMethod
+    public void isSignedInAsync(Promise promise) {
+        promise.resolve(GoogleSignIn.getLastSignedInAccount(getApplicationContext()) != null);
+    }
+
+    private void handleSignOutOrRevokeAccessTask(@NonNull Task<Void> task, final Promise promise) {
+        if (task.isSuccessful()) {
+            promise.resolve(true);
+        } else {
+            int code = getExceptionCode(task);
+            promise.reject(String.valueOf(code), GoogleSignInStatusCodes.getStatusCodeString(code));
+        }
+    }
+
+    private boolean addUIBlockOrReject(Runnable block, Promise promise) {
+        UIManager manager = mModuleRegistry.getModule(UIManager.class);
+        if (manager == null) {
+            promise.reject(new IllegalStateException("Implementation of " + UIManager.class.getName() + " is null. Are you sure you've included a proper Expo adapter for your platform?"));
+            return false;
+        } else {
+            manager.runOnUiQueueThread(block);
+            return true;
+        }
+    }
+
+    private GoogleSignInClient getClientOrReject(Promise promise) {
+        if (_apiClient == null) {
+            promise.reject(ERROR_EXCEPTION, "apiClient is null - call configure first");
+        }
+        return _apiClient;
+    }
+
+    private static class AccessTokenRetrievalTask extends AsyncTask<Bundle, Void, Bundle> {
+
+        private WeakReference<GoogleSignInModule> weakModuleRef;
+
+        AccessTokenRetrievalTask(GoogleSignInModule module) {
+            this.weakModuleRef = new WeakReference<>(module);
+        }
+
+        @Override
+        protected Bundle doInBackground(Bundle... params) {
+            Bundle user = params[0];
+            String mail = user.getString("email");
+            GoogleSignInModule moduleInstance = weakModuleRef.get();
+            if (moduleInstance == null) {
+                return user;
+            }
+            try {
+                ArrayList<String> scopes = user.getStringArrayList("scopes");
+                String token = GoogleAuthUtil.getToken(moduleInstance.getApplicationContext(), new Account(mail, "com.google"), scopesToString(scopes));
+                Bundle auth = user.getBundle("auth");
+                auth.putString("accessToken", token);
+                user.putBundle("auth", auth);
+                return user;
+            } catch (Exception e) {
+                moduleInstance.authTask.reject(ERROR_EXCEPTION, e.getLocalizedMessage());
+                return null;
+            }
+        }
+
+        @Override
+        protected void onPostExecute(Bundle result) {
+            super.onPostExecute(result);
+            GoogleSignInModule moduleInstance = weakModuleRef.get();
+            if (moduleInstance != null && result != null) {
+                moduleInstance.authTask.resolve(result);
+            }
+        }
+    }
+
+    private class EXGoogleSigninActivityEventListener implements ActivityEventListener {
```

Comment:
another nit - can you get rid of the `EX` prefix here? that's not needed in java because of package namespacing